packages/arb-compiler-evm/contract-templates/contracts/ArbSys.sol

```diff
@@ -28,4 +28,10 @@ interface ArbSys {
 
     // Send given amount of Eth to dest with from sender.
     function withdrawEth(address dest, uint256 amount) external;
+
+    // Return time when current message was posted on-chain
+    function currentMessageTime() external returns(uint);
```

Comment:
I think `messageTime` or `timeDelivered` might be better here. The word current is a bit weird. It's true that it's the current message we're processing, but it's kind of redundant in Ethereum's model of always being in a transaction context

Reply:
Also `timeDelivered` has the advantage of matching `timeUpperBound`. It'd be nice to have a uniform naming scheme

Reply:
I prefer not to use "delivered" because it's not when the message was delivered to the contract, it's when the message was submitted to the pending inbox.  We could go with messageTime.  Or maybe messageSubmissionTime?

Reply:
Actually, this is a non-mutating operation, so it could be invoked in the context of a non-mutating call.  Not clear what it should do in that case.  (And this might affect how we name it.)

Reply:
In a non-mutating call currently we just use the current time as the time the message is marked with. I also use a timebounds where the upper and lower bound are just the current time.

Reply:
So this, as currently implemented, will return the current time if invoked by/during a call?

Reply:
Yep. All three time values should be equal for a call. We can do that since we know exactly when it's executed. At least that's what makes sense to me currently